session/session.go

```diff
@@ -1089,9 +1089,12 @@ func (s *session) ParseSQL(ctx context.Context, sql, charset, collation string)
 		defer span1.Finish()
 	}
 	defer trace.StartRegion(ctx, "ParseSQL").End()
-	s.parser.SetSQLMode(s.sessionVars.SQLMode)
-	s.parser.SetParserConfig(s.sessionVars.BuildParserConfig())
-	return s.parser.Parse(sql, charset, collation)
+
+	p := s.parserPool.Get().(*parser.Parser)
+	defer s.parserPool.Put(p)
+	p.SetSQLMode(s.sessionVars.SQLMode)
```

Comment:
When the pool object is reused, the `SQLMode` will change the behaviour.

Reply:
Yes, so we have to set its SQL Mode each time before using it.